environs/config/config.go

```diff
@@ -595,6 +601,23 @@ func Validate(cfg, old *Config) error {
 			return fmt.Errorf("Invalid value for container-networking-method - %v", v)
 		}
 	}
+
+	if v, ok := cfg.defined[CloudInitUserDataKey].(string); ok && v != "" {
+		userDataMap, err := cfg.CloudInitUserData()
+		if err != nil {
+			return err
+		}
+		// error if users is specified
+		if _, ok := userDataMap["users"]; ok {
+			return errors.New("cloudinit-userdata: users not allowed")
+		}
+
+		// error if runcmd is specified
```

Comment:
also need to validate that "packages" is a list of strings

Reply:
done